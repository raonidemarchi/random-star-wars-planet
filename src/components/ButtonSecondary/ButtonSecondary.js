import React from 'react';
import styles from './ButtonSecondary.module.scss';

function ButtonSecondary({ children, onClick, disabled }) {
  return (
    <div role="button" className={[styles.button, disabled && styles.disabled].join(' ')} onClick={disabled ? () => {} : onClick}>
      {children}
    </div>
  )
}

export default ButtonSecondary;
