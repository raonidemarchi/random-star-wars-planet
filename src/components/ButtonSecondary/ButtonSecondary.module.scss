.button {
  padding: 16px 24px;
  min-width: 56px;
  border: 2px solid #f9a825;
  color: #f9a825;
  text-transform: uppercase;
  border-radius: 2px;
  text-align: center;
  letter-spacing: 0.04rem;
  font-weight: 700;
  cursor: pointer;
  user-select: none;

  &.disabled {
    opacity: .42;
    cursor: initial;
  }
}