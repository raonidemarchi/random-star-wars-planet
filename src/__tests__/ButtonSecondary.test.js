import React from 'react';
import ButtonSecondary from '../components/ButtonSecondary/ButtonSecondary';
import TestRenderer from 'react-test-renderer';

describe('ButtonSecondary component', () => {
  test('Should render a hello button', () => {
    const tree = TestRenderer.create(
      <ButtonSecondary disabled={false} onClick={() => 'hello'}>
        Hello
      </ButtonSecondary>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should execute onClick function', () => {
    const message = 'I should the value returned from onClick';
    const tree = TestRenderer.create(
      <ButtonSecondary onClick={() => message} />
    ).toJSON();

    expect(tree.props.onClick()).toBe(message);
  });

  test('Should not trigger on click function', () => {
    const tree = TestRenderer.create(
      <ButtonSecondary disabled={true} onClick={() => `this text shouldn't be returned`}>
        Disabled
      </ButtonSecondary>
    ).toJSON();

    expect(typeof tree.props.onClick()).toBe('undefined');
  });
});