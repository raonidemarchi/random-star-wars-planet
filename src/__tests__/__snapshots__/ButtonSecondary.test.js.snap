// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonSecondary component Should render a hello button 1`] = `
<div
  className="button false"
  onClick={[Function]}
  role="button"
>
  Hello
</div>
`;
